src/components/RecordMoviesComponent/index.tsx

```diff
@@ -0,0 +1,146 @@
+    const {
+        movieUrl,
+        movieTitle,
+        cleanupMovieState,
+        startRecording,
+        stopRecording,
+    } = props;
+    const [isRecording, setIsRecording] = useState<boolean>(false);
+    const [modalVisible, setModalVisible] = useState(false);
+    const [isHovering, setIsHovering] = useState(false);
+
+    const supportedBrowser = "VideoEncoder" in window;
+
+    const closeModal = () => {
+        cleanupMovieState();
+        setModalVisible(false);
+    };
+
+    const start = () => {
+        setIsRecording(true);
+        startRecording();
+    };
+
+    const stop = () => {
+        setIsRecording(false);
+        stopRecording();
+        setModalVisible(true);
+    };
+
+    const downloadMovie = () => {
+        if (!movieUrl) {
+            console.error("No recorded movie to download");
+            return;
+        }
+        const anchor = document.createElement("a");
+        anchor.href = movieUrl;
+        anchor.download = movieTitle;
+        document.body.appendChild(anchor);
+        anchor.click();
+        document.body.removeChild(anchor);
+        cleanupMovieState();
+    };
+
+    const startRecordIcon = (
+        <span className={styles.iconContainer}>
+            <span
+                className={classNames("icon-moon", "record-icon-circle")}
+            ></span>
+            <span
+                className={classNames("icon-moon", "record-icon-ring")}
+            ></span>
+        </span>
+    );
+
+    const getTooltipText = () => {
+        if (!supportedBrowser) {
+            return "Feature not supported on this browser, please try using Chrome or Edge";
+        } else if (isRecording) {
+            return "Stop movie recording";
+        } else {
+            return "Start movie recording";
+        }
+    };
+
+    const getIcon = () => {
+        const stopIcon = isHovering ? (
+            "stop-record-icon"
+        ) : (
+            <div
+                className={classNames([styles.redCircle, styles.animate])}
+            ></div>
+        );
+        return isRecording ? stopIcon : startRecordIcon;
+    };
+
+    const useFormattedDuration = () => {
+        const [formattedDuration, setFormattedDuration] = useState("");
+
+        useEffect(() => {
+            const video = document.createElement("video");
+            video.src = movieUrl;
+            video.onloadedmetadata = () => {
+                const durationInSeconds = video.duration;
+                const minutes = Math.floor((durationInSeconds % 3600) / 60);
+                const secs = Math.ceil(durationInSeconds) % 60;
+
+                const minutesStr = minutes.toString().padStart(2, "0");
+                const secondsStr = secs.toString().padStart(2, "0");
+
+                setFormattedDuration(`${minutesStr}:${secondsStr}`);
+                video.src = "";
+                video.load();
+            };
+        }, [movieUrl]);
+
+        return formattedDuration;
+    };
```

Comment:
`onloadedmetadata` is async hence the need for a hook, Promise or callback. Code was less complicated if this is handled up in `ViewerPanel` and store in state there but that file is already very cluttered with many concerns...

Reply:
could you just move this out into its own file?